crypto/merklesignature/keysBuilder.go

```diff
@@ -92,5 +92,5 @@ func generateKeysForRange(ctx context.Context, startIdx uint64, endIdx uint64, k
 		}
 		keys[k] = *sigAlgo
 	}
-	return nil
+	return nil //nolint:nilerr // OK to not return ctx.Err()
```

Comment:
Yes. (Also played with it a bit)
maybe for clarity, we can change line 87:
`return nil // we don't need to return the ctx error, since the other goroutine will report it. `